pkg/lib/cockpit-components-dialog.jsx

```diff
@@ -302,20 +308,24 @@
     const dialogObj = { };
     let error = null;
     dialogObj.props = props;
+    dialogObj.props.set_error = e => {
+        error = typeof e === 'object' && e !== null ? (e.message || e.toString()) : e;
+        dialogObj.render();
+    };
     dialogObj.footerProps = null;
     dialogObj.render = function() {
-        dialogObj.props.footer = <DialogFooter {...dialogObj.footerProps} />;
+        dialogObj.props.renderFooter = footerProps => <DialogFooter {...footerProps} />;
         // Don't render if we are no longer part of the document.
         // This would be mostly harmless except that it will remove
         // the input focus from whatever element has it, which is
         // unpleasant and also disrupts the tests.
         if (rootElement.offsetParent)
-            root.render(<Dialog {...dialogObj.props} error={error} />);
+            root.render(<Dialog {...dialogObj.props} footerProps={dialogObj.footerProps} error={error} />);
     };
     function updateFooterAndRender() {
         if (dialogObj.props === null || dialogObj.props === undefined)
             dialogObj.props = { };
-        dialogObj.props.footer = <DialogFooter {...dialogObj.footerProps} />;
+        dialogObj.props.renderFooter = footerProps => <DialogFooter {...footerProps} />;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19200-20230821-124816-2fdc0a03-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)